src/Plugins/OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/ComputeKernelAvgMisorientations.cpp

```diff
@@ -96,34 +96,35 @@ class FindKernelAvgMisorientationsImpl
             uint32_t phase1 = crystalStructures[cellPhases[point]];
             for(int32_t j = -kernelSize[2]; j < kernelSize[2] + 1; j++)
             {
-              size_t jStride = j * xPoints * yPoints;
+
               if(plane + j < 0 || plane + j > zPoints - 1)
               {
                 continue;
               }
+              const int64_t jStride = j * xPoints * yPoints;
               for(int32_t k = -kernelSize[1]; k < kernelSize[1] + 1; k++)
               {
-                size_t kStride = k * xPoints;
                 if(row + k < 0 || row + k > yPoints - 1)
                 {
                   continue;
                 }
+                const int64_t kStride = k * xPoints;
                 for(int32_t l = -kernelSize[0]; l < kernelSize[0] + 1; l++)
                 {
                   if(col + l < 0 || col + l > xPoints - 1)
                   {
                     continue;
                   }
-                  const size_t neighbor = point + (jStride) + (kStride) + (l);
-                  if(featureIds[point] == featureIds[neighbor])
+                  const int64_t neighbor = static_cast<int64_t>(point) + jStride + kStride + l;
+                  if(neighbor >= 0 && featureIds[point] == featureIds[static_cast<size_t>(neighbor)])
```

Comment:
My only comment is whether we should report an error if `neighbor` is less than 0.

Reply:
I don't think so. If the neighbor is 0 then that means that we ended up grabbing a voxel that we should not be using anyways so ignoring it is fine for the calculation.

It might be interesting for us as developers but it would not add any useful information to report it to the user.